lib/llm_composer/provider_runner.ex

```diff
@@ -0,0 +1,132 @@
+defmodule LlmComposer.ProviderRunner do
+  @moduledoc """
+  Handles provider execution logic including fallback strategies, routing,
+  and error handling for multiple provider configurations.
+  """
+
+  alias LlmComposer.LlmResponse
+  alias LlmComposer.Message
+  alias LlmComposer.Settings
+
+  require Logger
+
+  @type messages :: [Message.t()]
+
+  @doc """
+  Runs provider execution with fallback support for multiple providers.
+  """
+  @spec run(messages(), Settings.t(), Message.t()) :: {:ok, any()} | {:error, atom()}
+  def run(messages, %Settings{providers: [{provider, provider_opts}]} = settings, system_msg) do
+    provider_opts = get_provider_opts(provider_opts, settings)
+    provider.run(messages, system_msg, provider_opts)
+  end
+
+  def run(messages, %Settings{providers: providers} = settings, system_msg)
+      when is_list(providers) and length(providers) > 1 do
+    router = get_provider_router()
+
+    if Process.whereis(router) == nil do
+      {:error, :provider_router_not_started}
+    else
+      do_run(router, providers, messages, system_msg, settings)
+    end
```

Comment:
```suggestion
    if Process.alive?(router) do
      do_run(router, providers, messages, system_msg, settings)
    else
      {:error, :provider_router_not_started}
    end
```

I think this should be equivalent?

Reply:
it acepts a `pid()` type, we do use an `:atom` name (module name) in here